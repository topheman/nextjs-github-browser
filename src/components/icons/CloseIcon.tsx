import React from "react";

import { getSvgProps } from "../../utils/svg";

export default function HamburgerIcon(
  props: React.SVGProps<SVGSVGElement>
): JSX.Element {
  return (
    <svg
      {...getSvgProps(
        {
          viewBox: "0 0 24 24",
          role: "presentation",
          "aria-label": "Close",
        },
        props
      )}
    >
      <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z" />
      <path fill="none" d="M0 0h24v24H0z" />
    </svg>
  );
}
