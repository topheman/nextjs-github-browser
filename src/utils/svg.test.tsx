/* eslint-disable tailwindcss/no-custom-classname */
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";
import CloseIcon from "../components/icons/CloseIcon";

describe("An icon component", () => {
  it("matches snapshot", () => {
    const { container } = render(<CloseIcon />);
    expect(container.querySelector("svg")).toMatchSnapshot();
  });

  it('sets aria-hidden="false" if ariaLabel prop is present', () => {
    const { container } = render(<CloseIcon aria-label="icon" />);
    expect(container.querySelector("svg")).toHaveAttribute(
      "aria-hidden",
      "false"
    );
    expect(container.querySelector("svg")).toHaveAttribute(
      "aria-label",
      "icon"
    );
  });

  it("respects the className prop", () => {
    const { container } = render(<CloseIcon className="foo" />);
    expect(container.querySelector("svg")).toHaveAttribute("class", "foo");
  });

  it("respects the fill prop", () => {
    const { container } = render(<CloseIcon fill="#f00" />);
    expect(container.querySelector("svg")).toHaveAttribute("fill", "#f00");
  });

  it("respects the style prop", () => {
    const { container } = render(
      <CloseIcon style={{ verticalAlign: "middle" }} />
    );
    expect(container.querySelector("svg")).toHaveStyle({
      verticalAlign: "middle",
    });
  });
});
