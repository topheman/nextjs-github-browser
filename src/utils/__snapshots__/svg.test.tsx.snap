// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`An icon component matches snapshot 1`] = `
<svg
  aria-hidden="false"
  aria-label="Close"
  fill="currentColor"
  role="presentation"
  style="display: inline-block; user-select: none; overflow: visible;"
  viewBox="0 0 24 24"
  width="1em"
>
  <path
    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
  />
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
</svg>
`;
